powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/TelemetryServiceBehavior.java

```diff
@@ -0,0 +1,112 @@
+import io.getlime.security.powerauth.app.server.database.RepositoryCatalogue;
+import io.getlime.security.powerauth.app.server.database.repository.ActivationRepository;
+import io.getlime.security.powerauth.app.server.service.exceptions.UnknownTelemetryReportNameException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Service for telemetry related reports.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+@Service
+public class TelemetryServiceBehavior {
+
+    private static final int DAYS_FOR_MAU = 30;
+
+    // Report names
+    private static final String CURRENT_MAU = "CURRENT_MAU";
+    private static final String USERS_IN_PAST_DAYS = "USERS_IN_PAST_DAYS";
+
+    // Parameter names
+    private static final String PARAM_APPLICATION = "application";
+    private static final String PARAM_DAYS = "days";
+
+    // Result attribute names
+    private static final String RESULT_USERS = "users";
+    private static final String RESULT_APPLICATION = "application";
+    private static final String RESULT_DAYS = "days";
+
+    private final RepositoryCatalogue repositoryCatalogue;
+
+    @Autowired
+    public TelemetryServiceBehavior(RepositoryCatalogue repositoryCatalogue) {
+        this.repositoryCatalogue = repositoryCatalogue;
+    }
+
+    public TelemetryReportResponse report(String reportName, Map<String, Object> parameters) throws UnknownTelemetryReportNameException {
+        switch (reportName.toUpperCase()) {
+            case CURRENT_MAU -> {
+                final Map<String, Object> updatedParameters = new HashMap<>(parameters);
+                updatedParameters.put(PARAM_DAYS, DAYS_FOR_MAU);
+                return reportUsersInPastDays(CURRENT_MAU, updatedParameters);
+            }
+            case USERS_IN_PAST_DAYS -> {
+                return reportUsersInPastDays(USERS_IN_PAST_DAYS, parameters);
+            }
+            default -> {
+                throw new UnknownTelemetryReportNameException("Unknown report name: " + reportName);
+            }
+        }
+    }
+
+    /**
+     * Prepare the users in past days report
+     * @param reportName Provided report name.
+     * @param parameters Report parameters.
+     * @return Number of unique app users in specified period.
+     */
+    private TelemetryReportResponse reportUsersInPastDays(String reportName, Map<String, Object> parameters) throws UnknownTelemetryReportNameException {
+        final Integer days = (Integer) parameters.get(PARAM_DAYS);
+        final String applicationId = (String) parameters.get(PARAM_APPLICATION);
+
+        // Validate report parameters are present
+        if (days == null) {
+            throw new UnknownTelemetryReportNameException("Missing parameter 'days'.");
```

Comment:
I would rather throw `jakarta.validation.ValidationException`, definitely not `UnknownTelemetryReportNameException`, or let's rename it to `TelemetryReportException`.

Reply:
Good catch, this is a glitch in naming, I will fix it.